app/assets/javascripts/discourse/app/components/topic-map/private-message-map.gjs

```diff
@@ -0,0 +1,257 @@
+      remove: this.toggleEditing,
+      add: this.args.showInvite,
+    };
+  }
+
+  get actionAllowedLabel() {
+    return `private_message_info.${this._actionAllowedKey()}`;
+  }
+
+  @action
+  toggleEditing() {
+    this.isEditing = !this.isEditing;
+  }
+
+  <template>
+    <div class={{this.participantsClasses}}>
+      {{#each @postAttrs.allowedGroups as |group|}}
+        <PmMapUserGroup
+          @model={{group}}
+          @isEditing={{this.isEditing}}
+          @canRemoveAllowedUsers={{@postAttrs.canRemoveAllowedUsers}}
+          @removeAllowedGroup={{@removeAllowedGroup}}
+        />
+      {{/each}}
+      {{#each @postAttrs.allowedUsers as |user|}}
+        <PmMapUser
+          @model={{user}}
+          @isEditing={{this.isEditing}}
+          @canRemoveAllowedUsers={{@postAttrs.canRemoveAllowedUsers}}
+          @canRemoveSelfId={{@postAttrs.canRemoveSelfId}}
+          @removeAllowedUser={{@removeAllowedUser}}
+        />
+      {{/each}}
+    </div>
+
+    {{#if this.canShowControls}}
+      <div class="controls">
+        <DButton
+          @action={{this.actionAllowed}}
+          @label={{this.actionAllowedLabel}}
+          class="btn btn-default add-remove-participant-btn"
+        />
+
+        {{#if (and this.canInvite this.isEditing)}}
+          <DButton
+            @action={{@showInvite}}
+            @icon="plus"
+            class="btn btn-default no-text btn-icon add-participant-btn"
+          />
+        {{/if}}
+      </div>
+    {{/if}}
+  </template>
+}
+
+class PmMapUserGroup extends Component {
+  get canRemoveLink() {
+    return this.args.isEditing && this.args.canRemoveAllowedUsers;
+  }
+
+  get groupUrl() {
+    return groupPath(this.args.model.name);
+  }
+
+  <template>
+    <div class="user group">
+      <a href={{this.groupUrl}} class="group-link">
+        {{dIcon "users"}}
+        <span class="group-name">{{@model.name}}</span>
+      </a>
+    </div>
+    {{#if this.canRemoveLink}}
+      <PmRemoveGroupLink
+        @model={{@model}}
+        @removeAllowedGroup={{@removeAllowedGroup}}
+      />
+    {{/if}}
+  </template>
+}
+
+class PmRemoveGroupLink extends Component {
+  @service dialog;
+
+  @action
+  showConfirmDialog() {
+    this.dialog.deleteConfirm({
+      message: I18n.t("private_message_info.remove_allowed_group", {
+        name: this.args.model.name,
+      }),
+      confirmButtonLabel: "private_message_info.remove_group",
+      didConfirm: () => this.args.removeAllowedGroup(this.args.model),
+    });
+  }
+
+  <template>
+    {{! template-lint-disable no-invalid-interactive }}
+    <a
+      class="remove-invited no-text btn-icon btn"
+      {{on "click" this.showConfirmDialog}}
+    >
+      {{dIcon "times"}}
+    </a>
+  </template>
+}
+
+class PmMapUser extends Component {
+  @service site;
+
+  get linkClass() {
+    if (this.site.mobileView) {
+      return "";
+    }
+    return "user-link";
+  }
+
+  get userUrl() {
+    return this.args.model.path;
+  }
```

Comment:
It's personal preference, but this logic is very simple so I think I would inline this logic in the template rather than abstracting it away. That way, people can just glance at the template rather than having to trace the getter.

But, this is not a blocker. I'll leave it up to you to decide.